pkg/template/controller/readiness.go

```diff
@@ -0,0 +1,99 @@
+package controller
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/apps"
+	"k8s.io/kubernetes/pkg/apis/batch"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	deploymentutil "k8s.io/kubernetes/pkg/controller/deployment/util"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/apis/apps"
+)
+
+func checkDeploymentReadiness(obj runtime.Object) (bool, bool) {
```

Comment:
answers from @smarterclayton, roughly interpreted:

this is fine for now for detection logic, but the follow up is to do an upstream pull to introduce the various resource checking logic as a utility function.

and longer term expose that logic as an api.